packages/core/react/use-chat.ts

```diff
@@ -320,6 +328,52 @@ const getStreamedResponse = async (
       responseMessage['function_call'] = parsedFunctionCall;
 
       mutate([...chatRequest.messages, { ...responseMessage }]);
+    } else {
+      // Check for an embedded function call.
+      // TODO: Support N function calls?
+      let functionCallIndex = streamedResponse.indexOf('{"function_call":');
+      if (functionCallIndex !== -1) {
+        let textResponse = streamedResponse.substring(0, functionCallIndex);
+        let functionCallString = streamedResponse.substring(functionCallIndex);
+
+        // Create a message for the text before the function call
+        if (textResponse) {
```

Comment:
I still have to find a way to repro this outside of my application, but there this `if` condition causes the message before the function_call to be duplicated. Without it, the solution works perfectly!

Reply:
Hi @woodbridge , I was able to repro here: https://github.com/StefanDjokovic/ai-chatbot-repro/pull/1 I linked the ai sdk and I see the behaviour of duplicated messages. NOTE: it's possible that there is a bug in the template itself; not sure if that's the case; if you also have a repro were this behaviour doesn't happen lmk and I'll look more into this.

<img width="617" alt="image" src="https://github.com/vercel/ai/assets/37225048/653eaf70-2b7e-4873-b93c-def70ad8d561">
<img width="384" alt="image" src="https://github.com/vercel/ai/assets/37225048/9e6d5b17-c374-4aff-a4d6-bb17da97fdd0">

Reply:
@woodbridge @StefanDjokovic has this been figured out / is it related directly to the changes made here?

Reply:
yeah believe it's a bug in my changes - havent' had time to find the bug yet, but have been using this fork internally

Reply:
I think we'll need to wait for that to be resolved before merging this. Thanks for your work here 🙏